### DISCUSSION TITLE
Underlying exception is sometimes swallowed in the logs

### DISCUSSION THREAD

**Discussion by changlinli** [awslabs/amazon-kinesis-video-streams-producer-sdk-java#180] [Ideas]
I occasionally see the following log message.

```
EXCEPTION: RuntimeException: Exception thrown on sending thread: Exception while sending encoded chunk in MKV stream !
```

which ultimately comes from https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/blob/026f890c7b8975e423c859062a807220e290bd23/src/main/java/com/amazonaws/kinesisvideo/client/PutMediaClient.java#L138. 

The previous line (https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/blob/026f890c7b8975e423c859062a807220e290bd23/src/main/java/com/amazonaws/kinesisvideo/client/PutMediaClient.java#L137) also logs the exception at a debug level, which results in

```
Exception while sending data.
```

Without the actual exception though, that debug message is fairly useless. The exception is in fact passed in as an argument to the `log.debug` statement, but it's never actually used, because `log.debug` expects a string template that looks something like `"My message: %s"`and no interpolation arguments are ever given in the log message.

Likewise nothing ever actually ends up printing the bottom level exception, because the top-level error handler (https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/blob/026f890c7b8975e423c859062a807220e290bd23/src/main/java/com/amazonaws/kinesisvideo/common/logging/Log.java#L289) only ever prints the outer-most exception.

So the actual exception that caused all of this is effectively swallowed and never shown in the logs.

**Comment by MushMal** [comment]
Just a note that we are encouraging folks to submit PRs. Please feel free to open a PR. The normal process is to make modifications, create a new branch and push it in. The output will have a link to the PR which can be pushed for a review.
